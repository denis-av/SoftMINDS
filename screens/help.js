import { StatusBar } from "expo-status-bar";
import React from "react";
import { Image, StyleSheet, Text, View, ImageBackground, Pressable } from "react-native";
import AppIntroSlider from "react-native-app-intro-slider";

const slides = [
  {
    key: "one",
    title: "HOW TO PLAY",
    text: "By navigating to the levels screen, you will be able to see all the levels. The levels with the faded out background are not able to be played until you solve de previous levels. After entering in a level screen you need to input the solution to the given problem.",
    image: require('../app/images/how_to_play_icon.png'),
  },
  {
    key: "two",
    title: "SCORING SYSTEM",
    text: "Above each given problem which consists of an image there is a timer, dependig on the time it takes toy to solve, you will recevie a certain score: 3 points under 30 seconds, 2 points under 1 minute, 1 point under 2 minutes, 0 points above 2 minutes.",
    image: require('../app/images/scoring_system_icon.png'),
  },
  {
    key: "three",
    title: "RANKING PROCEDURE",
    text: "Into the ranking screen you will find the best players together with the total score obtained by them. There are 3 types of tiers: newbie (under 12 points), intermediate (starting from 12 points to 23), master (starting from 24 points).",
    image: require('../app/images/ranking_procedure_icon.png'),
  },
];

export default class Help extends React.Component {

  state = { showHomePage: false };
  _renderItem = ({ item }) => {
    return (
      <View style={{backgroundColor:'#1a1a1a'}}>
        <Image source={item.image} style={{ resizeMode: "cover", alignSelf:'center' }}/>
        <Text style={{ paddingTop: 25, paddingBottom: 10,fontFamily:"bold-font", fontSize: 37, fontWeight: "bold", color: "#F3A416", alignSelf: "center", marginTop:'5%'}}>{item.title}</Text>
        <Text style={{ textAlign:"center", color:"#fff", fontSize:19, paddingHorizontal:30, marginTop:'7%', fontFamily:"normal-font" }}>{item.text}</Text>
      </View>
    );
  };

  render() {
    if (this.state.showHomePage){
      return <Help/>
    } else 
    return ( 
          <View style={{flex: 1}}>
            <ImageBackground source={require("../app/images/bg4.png")} style={{width:"100%", height:"100%"}}>
              <View style={{flex: 0.13, backgroundColor:"#F3A416", flexDirection:'row', alignItems:'flex-end'}}>
                      <Pressable style={{flex: 0.25, height:"75%", alignItems:'center', justifyContent:'center'}} onPress={() => this.props.navigation.navigate("Menu")}>
                          <Image source={require("../app/images/back-2.png")} resizeMode = "center" style={{width:"35%", height:"35%"}}/>
                      </Pressable>
                  <View style={{flex: 0.50, height:"75%", alignItems:'center', justifyContent:'center'}}>
                      <Text style={{fontSize: 40, color:"white", fontFamily:"bold-font"}}>HELP</Text>
                  </View>
            </View>
            <View style={{flex:0.87, width:"100%",height:"65%", marginTop:'25%'}}>
              <AppIntroSlider renderItem={this._renderItem} data={slides} activeDotStyle={{ backgroundColor:"#F3A416" }} dotStyle={{ backgroundColor:"#875f18"}} nextLabel="" doneLabel=""/>
            </View>
          </ImageBackground>
        </View>
    );
  }
}



