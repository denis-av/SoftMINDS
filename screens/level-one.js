import React from 'react';
import {View, ImageBackground, Pressable, Text, Image, TextInput} from 'react-native';
import NumberCard from '../components/number-card';

export default class LevelOne extends React.Component{
    constructor(){
        super();
        this.state = {
            answer: ""
        };
    };

    handleAnswer = (text) => {this.setState({answer: text})};

    render(){
        return(
            <View style={{flex:1}}>
                <ImageBackground source={require("../app/images/bg4.png")} style={{width:"100%", height:"100%"}}>
                    <View style={{flex: 0.13, backgroundColor:"#F3A416", flexDirection:'row', alignItems:'flex-end'}}>
                            <Pressable style={{flex: 0.25, height:"75%", alignItems:'center', justifyContent:'center'}} onPress={() => this.props.navigation.navigate("Levels")}>
                                <Image source={require("../app/images/back-2.png")} resizeMode = "center" style={{width:"35%", height:"35%"}}/>
                            </Pressable>
                        <View style={{flex: 0.50, height:"75%", alignItems:'center', justifyContent:'center'}}>
                            <Text style={{fontSize: 40, color:"white", fontFamily:"bold-font"}}>LEVEL 1</Text>
                        </View>
                    </View>
                    <View style={{flex: 0.09, alignItems:'center', justifyContent:'center'}}>
                        <Text style={{fontSize:35, color:'white', fontFamily:'bold-font'}}>00:10</Text>
                    </View>
                    <View style={{flex: 0.50, backgroundColor:'blue'}}>

                    </View>
                    <View style={{flex: 0.10,  flexDirection:'row', alignItems:'flex-end', justifyContent: 'space-around'}}>
                        <View style={{flexDirection:'row', alignItems:'center', borderWidth:2, borderTopWidth:0, borderLeftWidth:0, borderRightWidth:0, height:"50%",borderColor:'white', width:"48%", marginLeft:"7%"}}>
                            <TextInput placeholder="Answer here" 
                                       placeholderTextColor="#696969"
                                       style={{color: '#696969',fontSize:21, fontFamily:'bold-font'}}
                                       value={this.state.answer}
                                       onChangeText={this.handleAnswer}
                            />
                            <Image source={require("../app/images/x.png")} resizeMode = "contain" style={{width:"40%", height:"40%", marginLeft:"14%"}}/>
                        </View>
                        
                        <Pressable style={{backgroundColor: '#F3A416', height:"70%", width:"31%", alignItems:'center', justifyContent:'center', marginRight:"7%"}}  onPress={() => console.log("aaa")} >
                            <Text style={{fontSize: 25, fontFamily:'bold-font', color:'white'}}>SUBMIT</Text>
                        </Pressable>
                    </View>
                    <View style={{flex: 0.22, justifyContent:'space-around', alignContent:'center', marginHorizontal:"7%"}}>
                        <View style={{flexDirection:'row', alignItems:'center', justifyContent:'space-around', marginTop:"5%"}}>
                            <NumberCard number={"0"} bgColor="#F3A416"/>
                            <NumberCard number={"1"} bgColor="#F3A416"/>
                            <NumberCard number={"2"} bgColor="#F3A416"/>
                            <NumberCard number={"3"} bgColor="#F3A416"/>
                            <NumberCard number={"4"} bgColor="#F3A416"/>
                        </View>
                        <View style={{flexDirection:'row', alignItems:'center', justifyContent:'space-around', marginBottom:"10%"}}>
                            <NumberCard number={"5"} bgColor="#F3A416"/>
                            <NumberCard number={"6"} bgColor="#F3A416"/>
                            <NumberCard number={"7"} bgColor="#F3A416"/>
                            <NumberCard number={"8"} bgColor="#F3A416"/>
                            <NumberCard number={"9"} bgColor="#F3A416"/>
                        </View>
                        
                    </View>
                </ImageBackground>
            </View>
        );
    }
}