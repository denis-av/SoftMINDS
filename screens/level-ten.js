import React from 'react';
import {View, ImageBackground, Pressable, Text, Image, TextInput, TouchableOpacity, Alert} from 'react-native';
import NumberCard from '../components/number-card';
import { Stopwatch} from 'react-native-stopwatch-timer';
import * as firebase from "firebase";
import AsyncStorage from '@react-native-async-storage/async-storage';
 
export default class LevelTen extends React.Component{
 
    #currentTime = "";
    #email = "";
 
    constructor(){
        super();
        this.state = {
            answer: "",
            isStopwatchActive: false,
            correctAnswer: '26',
            userScore: 0,
        };
 
    };
 
    componentDidMount(){
        this.setState({isStopwatchActive:true});
        this.getCurrentUser();
        this.getCurrentUserPoints();
    }

    getCurrentUserPoints = () => {
        firebase.database().ref('/ranking').once('value').then((snapshot) => {
            var tempTotalScore = 0;
            snapshot.forEach( (childSnapshot) => {
                if(childSnapshot.val().username === this.#email.split("@")[0].replace('.','').replace('_','')){
                    tempTotalScore = childSnapshot.val().score;
                }
            })
            this.setState({userScore: tempTotalScore});
        })
    }

    getCurrentUser = async() => {
        try{
            const value = await AsyncStorage.getItem("email");
            if(value !== null) {
                this.#email = value;
            }
        }catch(e){ }
    }
 
    handleAnswer = () =>{
        const username = this.#email.split("@")[0].replace('.','').replace('_','');
        if( this.state.answer === this.state.correctAnswer ){
            this.setState({isStopwatchActive:!this.state.isStopwatchActive})
            firebase.database().ref("/users").child(username).set({
                username: username,
                levelOne: "true",
                levelTwo: "true",
                levelThree: "true",
                levelFour: "true",
                levelFive: "true",
                levelSix: "true",
                levelSeven: "true",
                levelEight: "true",
                levelNine: "true",
                levelTen: "true",
                levelEleven: "",
                levelTwelve: ""
            })

            if( this.#currentTime.localeCompare("00:30") < 0 ){
                points = 3;
            }else if( (this.#currentTime.localeCompare("00:30") > 0) && (this.#currentTime.localeCompare("01:00") < 0)){
                points = 2;
            }else if((this.#currentTime.localeCompare("01:00") > 0) && (this.#currentTime.localeCompare("02:00") < 0)){
                points = 1;
            }else if(this.#currentTime.localeCompare("02:00") >0 ){
                points = 0;
            }

            firebase.database().ref("/ranking").child(username).update({
                username: username,
                score: points + this.state.userScore,
            })

            Alert.alert(
                "Correct Answer",
                "You submitted the correct answer!",
                [
                  { text: "Go back to levels", onPress: () => {this.props.navigation.reset({index:0, routes:[{name:'Levels'}]}) }}
                ]
              );
        } else{
            Alert.alert(
                "Wrong Answer",
                "The answer is incorrect!",
                [
                  { text: "Try again", onPress: () => this.setState({answer:""})}
                ]
              );
        }
    }
 
    handleStopwatch(){
        console.log(this.formatTime);
    }
 
    render(){
        return(
            <View style={{flex:1}}>
                <ImageBackground source={require("../app/images/bg4.png")} style={{width:"100%", height:"100%"}}>
                    <View style={{flex: 0.13, backgroundColor:"#F3A416", flexDirection:'row', alignItems:'flex-end'}}>
                            <Pressable style={{flex: 0.25, height:"75%", alignItems:'center', justifyContent:'center'}} onPress={() => this.props.navigation.navigate("Levels")}>
                                <Image source={require("../app/images/back-2.png")} resizeMode = "center" style={{width:"35%", height:"35%"}}/>
                            </Pressable>
                        <View style={{flex: 0.50, height:"75%", alignItems:'center', justifyContent:'center'}}>
                            <Text style={{fontSize: 40, color:"white", fontFamily:"bold-font"}}>LEVEL 10</Text>
                        </View>
                    </View>
                    <View style={{flex: 0.09, alignItems:'center', justifyContent:'center'}}>
                        {/* <Text style={{fontSize:35, color:'white', fontFamily:'bold-font'}}>00:10</Text> */}
                        <Stopwatch start={this.state.isStopwatchActive} options={{container: {
    backgroundColor: 'transparent',
    borderRadius: 5,
 
  },
  text: {
    fontSize:35, color:'white', fontFamily:'bold-font'
  }}}
  getTime={(time)=> this.#currentTime = time}
 
  />
                    </View>
                    <View style={{flex: 0.50, alignItems:'center', justifyContent:'center'}}>
                        <Image source={require("../app/images/level_10.png")} resizeMode = "contain" style ={{width:"100%", height:"100%"}}/>
                    </View>
                    <View style={{flex: 0.10,  flexDirection:'row', marginHorizontal:"7%", alignItems:'center', justifyContent:'center'}}>
                        <View style={{flex: 0.60, flexDirection:'column', justifyContent:'flex-end', height:"70%"}}>
                            <View style={{flexDirection:'row', alignItems:'center'}}>
                                <TextInput placeholder="Answer here" 
                                       placeholderTextColor="#696969"
                                       style={{color: '#696969',fontSize:23, fontFamily:'bold-font', width:"85%"}}
                                       value={this.state.answer}
                                       editable = {false}
                                />
                                <TouchableOpacity style ={{width:30, height:30, alignItems:'center', justifyContent:'center'}} onPress={() => this.setState({answer: ""})}>
                                    <Image source={require("../app/images/x.png")} resizeMode = "contain" style={{width:"70%", height:"70%"}}/>
                                </TouchableOpacity>
                            </View>
                        <View style={{flexDirection:'row', alignItems:'center', borderWidth:2, borderTopWidth:0, borderLeftWidth:0, borderRightWidth:0,borderColor:'white', width:"97%"}}></View>
                        </View>
                            <View style={{flex:0.40}}>
                                <Pressable style={{backgroundColor: '#F3A416', height:"70%", alignItems:'center', justifyContent:'center', marginLeft:"7%"}}  onPress={this.handleAnswer} >
                                    <Text style={{fontSize: 25, fontFamily:'bold-font', color:'white'}}>SUBMIT</Text>
                                </Pressable>
                            </View>
                    </View>
                    <View style={{flex: 0.22, justifyContent:'space-evenly',marginBottom:"7%", alignContent:'center', marginHorizontal:"7%"}}>
                        <View style={{flexDirection:'row', alignItems:'center', justifyContent:'space-between'}}>
                            <NumberCard number={"0"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("0")})}/>
                            <NumberCard number={"1"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("1")})}/>
                            <NumberCard number={"2"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("2")})}/>
                            <NumberCard number={"3"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("3")})}/>
                            <NumberCard number={"4"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("4")})}/>
                        </View>
                        <View style={{flexDirection:'row', alignItems:'center', justifyContent:'space-between'}}>
                            <NumberCard number={"5"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("5")})}/>
                            <NumberCard number={"6"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("6")})}/>
                            <NumberCard number={"7"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("7")})}/>
                            <NumberCard number={"8"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("8")})}/>
                            <NumberCard number={"9"} bgColor="#F3A416" onPress = {() => this.setState({answer: this.state.answer.concat("9")})}/>
                        </View>
 
                    </View>
                </ImageBackground>
            </View>
        );
    }
}