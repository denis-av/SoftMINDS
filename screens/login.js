import React from 'react';
import {View, TouchableOpacity, Text,Pressable,Image, ImageBackground, TextInput, StatusBar} from 'react-native';
import * as firebase from "firebase";
import AsyncStorage from '@react-native-async-storage/async-storage';

export default class Login extends React.Component{
    constructor(){
        super();
        this.state = {
            email: "",
            password: "",
            error : ""
        };
    }

    handleEmail = (text) => {this.setState({email: text})};
    handlePassword = (text) => {this.setState({password: text})};

    storeData = async() => {
        try{
            await AsyncStorage.setItem("email",this.state.email)
        }catch(e){ }
    }

    LoginPress = () => {
        firebase.auth().signInWithEmailAndPassword(this.state.email,this.state.password)
        .then(() => {
            this.props.navigation.navigate('Menu');
        })
        .catch(error => this.setState ({error : error.message}))
        this.storeData();
    }


    render(){
        return(
            <View style={{flex: 1, flexDirection: 'column', alignItems: 'center', justifyContent: 'center'}}>
               <StatusBar barStyle={"light-content"}/>
               <ImageBackground  source={require("../app/images/bg3.jpeg")} style={{width: "100%", height: "100%"}}> 
                 <View style={{flex: 0.3, flexDirection: 'row', marginTop: 70, alignItems:'center'}}>
                    <Image source={require("../app/images/logo1.png")} style={{width: "33%", height: "33%"}} resizeMode='contain'/>
                    <Text style={{color:'white',fontFamily:'bold-font', fontSize:59, marginTop:"17%",marginLeft:"-13%"}}>SoftMINDS</Text>
                 </View>
                 <View style={{flex: 0.30, flexDirection: 'column', alignItems:'center', justifyContent:'center'}}>
                    <View style={{flex: 0.33, width:"80%", marginHorizontal:"10%", justifyContent:'center'}}>
                            <View style={{flexDirection:'row', alignItems:'center', borderWidth:2, borderTopWidth:0, borderLeftWidth:0, borderRightWidth:0, height:"50%",borderColor:'white'}}>
                                <Image source={require("../app/images/email.png")} resizeMode='contain' style={{width:"45%", height:"45%", marginLeft:"-17%"}}/>
                                <TextInput placeholder="Email" 
                                        placeholderTextColor="white"
                                        style={{color: 'white', marginLeft:"-11%",fontSize:19, fontFamily:'bold-font'}}
                                        value={this.state.email}
                                        onChangeText={this.handleEmail}
                                />
                            </View>
                        </View>
                        <View style={{flex: 0.33, width:"80%", marginHorizontal:"10%", justifyContent:'center'}}>
                            <View style={{flexDirection:'row', alignItems:'center', borderWidth:2, borderTopWidth:0, borderLeftWidth:0, borderRightWidth:0, height:"50%",borderColor:'white'}}>
                                <Image source={require("../app/images/password.png")} resizeMode='contain' style={{width:"60%", height:"60%", marginLeft:"-24%"}}/>
                                <TextInput placeholder="Password" 
                                        placeholderTextColor="white"
                                        secureTextEntry={true}
                                        style={{color: 'white', marginLeft:"-19%",fontSize:19, fontFamily:'bold-font'}}
                                        value={this.state.password}
                                        onChangeText={this.handlePassword}
                                />
                            </View>
                        </View>
                 </View>
                 <View style={{flex:0.10, justifyContent:'center'}}>
                    <Pressable style={{backgroundColor: '#F3A416', alignItems:'center', width:"50%", marginHorizontal:"25%", height:47, justifyContent:'center', borderRadius:30}}
                               onPress={this.LoginPress} > 
                        <Text style={{color:'white', fontFamily:'bold-font', fontSize:17}}>LOGIN</Text>
                    </Pressable>
                 </View>
                 <View style={{flex:0.10, width:"50%", alignItems:'center', justifyContent:'center', marginHorizontal:"25%"}}>
                    <TouchableOpacity style={{justifyContent:'center', alignItems:'center'}} onPress={() => this.props.navigation.navigate("SignUP")}>
                            <Text style={{color:'white', fontSize:16, fontFamily:'normal-font'}}>
                                Don't have an account?
                            </Text>
                            <Text style={{color:'white', fontSize:17, fontFamily:'bold-font'}}>
                                REGISTER
                            </Text>
                    </TouchableOpacity>
                </View>
               </ImageBackground> 
            </View>
        );
    }
}