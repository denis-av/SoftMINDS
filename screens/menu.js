import React from 'react';
import {View, TouchableOpacity, Text,Pressable,Image, ImageBackground, TextInput, StatusBar, TouchableHighlight} from 'react-native';


export default class Menu extends React.Component{
    constructor(){
        super();
        this.state = {};
    }

    render(){
        return(
            <View style={{flex: 1, flexDirection: 'column', alignItems: 'center', justifyContent: 'center'}}>
                <StatusBar barStyle={"light-content"}/>
                <ImageBackground  source={require("../app/images/bg2.jpeg")} style={{width: "100%", height: "100%"}}> 
                <View style={{flex:0.10, width:"30%", justifyContent:'flex-end', marginHorizontal:"25%", marginLeft:"-5%", marginTop:"3%"}}>
                    <TouchableOpacity style={{justifyContent:'center', alignItems:'center'}} onPress={() => this.props.navigation.navigate("Login")}>
                            <Image source={require("../app/images/logout_icon.png")} style={{width: "57%", height: "57%"}} resizeMode='contain'/>
                    </TouchableOpacity>
                </View>
                 <View style={{flex: 0.80, width:"74%", flexDirection: 'column', alignItems:'center', marginHorizontal:"13%", justifyContent:'center'}}>
                     <View style={{width:'100%', height:'60%', justifyContent:'space-around', alignItems:'center', marginBottom:'10%'}}>
                             <TouchableOpacity style={{backgroundColor:'#F3A416', width:"80%",height:70, borderRadius:20}} onPress={()=>this.props.navigation.navigate("Levels")}>
                                <View style={{flexDirection:'row',alignItems:'center',justifyContent:'flex-start', width:"100%",height:70, borderRadius:20, overflow:'hidden'}}>
                                    <Image source={require("../app/images/levels_icon.png")} style={{height:35,width:35, marginLeft:'15%'}} resizeMode='center'/>
                                    <Text style={{color:'white', fontFamily:'bold-font', fontSize:27,marginLeft:'10%'}}>LEVELS</Text>
                                </View>
                            </TouchableOpacity>
                            <TouchableOpacity style={{backgroundColor:'#F3A416', width:"80%",height:70, borderRadius:20}}>
                                <View style={{flexDirection:'row',alignItems:'center',justifyContent:'flex-start', width:"100%",height:70, borderRadius:20, overflow:'hidden'}}>
                                    <Image source={require("../app/images/ranking_icon.png")} style={{height:35,width:35, marginLeft:'15%'}} resizeMode='center'/>
                                    <Text style={{color:'white', fontFamily:'bold-font', fontSize:27, marginLeft:'10%'}}>RANKING</Text>
                                </View>
                            </TouchableOpacity>
                            <TouchableOpacity style={{backgroundColor:'#F3A416', width:"80%",height:70, borderRadius:20}}>
                                <View style={{flexDirection:'row',alignItems:'center',justifyContent:'flex-start', width:"100%",height:70, borderRadius:20, overflow:'hidden'}}>
                                    <Image source={require("../app/images/brain_icon.png")} style={{height:35,width:35, marginLeft:'15%'}} resizeMode='center'/>
                                    <Text style={{color:'white', fontFamily:'bold-font', fontSize:27, marginLeft:'10%'}}>TRAINING</Text>
                                </View>
                            </TouchableOpacity>
                            <TouchableOpacity style={{backgroundColor:'#F3A416', width:"80%",height:70, borderRadius:20}}  onPress={()=>this.props.navigation.navigate("Help")}>
                                <View style={{flexDirection:'row',alignItems:'center',justifyContent:'flex-start', width:"100%",height:70, borderRadius:20, overflow:'hidden'}}>
                                    <Image source={require("../app/images/help_icon.png")} style={{height:35,width:35, marginLeft:'15%'}} resizeMode='center'/>
                                    <Text style={{color:'white', fontFamily:'bold-font', fontSize:27, marginLeft:'10%'}}>HELP</Text>
                                </View>
                            </TouchableOpacity>
                     </View>
                 </View>
                 <View style={{flex: 0.2, flexDirection: 'row', marginTop:"-20%", alignItems:'center'}}>
                    <Image source={require("../app/images/logo1.png")} style={{width: "33%", height: "33%", marginTop:"-30%", marginLeft:"17%"}} resizeMode='contain'/>
                    <Text style={{color:'white',fontFamily:'bold-font', fontSize:30, marginTop:"-21%",marginLeft:"-14%"}}>SoftMINDS</Text>
                 </View>
                </ImageBackground> 
             </View>
        );
    }
}