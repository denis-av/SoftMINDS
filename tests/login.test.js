import React from 'react';
import Login from '../screens/login';
import {findByTestAttr} from '../screens/login';
import * as firebase from 'firebase';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const navigation = {navigate: jest.fn()};

jest.mock('firebase', () => {
    return {
      auth: jest.fn().mockReturnThis(),
      signInWithEmailAndPassword: jest.fn().mockImplementation(() => Promise.resolve()),
    };
  });


test('renders correctly the snapshot', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

describe('Login', () => {
    it('show SoftMINDS text from button', () => {
        const wrapper = shallow(<Login />);
        const texts = wrapper.find('Text').at(0);
        expect(texts.props().children).toBe('SoftMINDS');
    })
})

describe('Login', () => {
    it('show LOGIN text from button', () => {
        const wrapper = shallow(<Login />);
        const texts = wrapper.find('Text').at(1);
        expect(texts.props().children).toBe('LOGIN');
    })
})

describe('Login', () => {
    it('show text1', () => {
      const wrapper = shallow(<Login />);
      const texts = wrapper.find('Text').at(2);
      expect(texts.props().children).toBe("Don't have an account?");
    })
  })
  
  describe('Login', () => {
    it('show REGISTER text', () => {
      const wrapper = shallow(<Login />);
      const texts = wrapper.find('Text').at(3);
      expect(texts.props().children).toBe('REGISTER');
    })
  })

  describe('Login', () => {
    it('check if the initial state of email is empty', () => {
      const wrapper = shallow(<Login />);
      const instance = wrapper.instance();
      jest.spyOn(instance, 'handleEmail');
      expect(instance.state.email).toBe("");
    })
  })
  
  describe('Login', () => {
    it('check if the initial state of password is empty', () => {
      const wrapper = shallow(<Login />);
      const instance = wrapper.instance();
      jest.spyOn(instance, 'handlePassword');
      expect(instance.state.email).toBe("");
    })
  })
  
  describe('Login', () => {
    it('check the updated state of the email', () => {
      const wrapper = shallow(<Login />);
      const instance = wrapper.instance();
      const input = findByTestAttr(wrapper, "testInputUsername");
      input.simulate("changeText", "example@gmail.com");
      wrapper.update();
      jest.spyOn(instance, 'handleEmail');
      expect(instance.state.email).toBe("example@gmail.com");
    })
  })
  
  describe('Login', () => {
    it('check if the password is not empty after the initialization', () => {
      const wrapper = shallow(<Login />);
      const instance = wrapper.instance();
      const input = findByTestAttr(wrapper, "testInputPassword");
      input.simulate("changeText", "123456");
      wrapper.update();
      jest.spyOn(instance, 'handlePassword');
      expect(instance.state.password).not.toBe("");
    })
  })
  
  describe('Login', () => {
    it('FireBase Test', () => {
        const navigate = jest.fn();
        const wrapper = shallow(<Login navigation = {navigation}/>);
        const instance = wrapper.instance();
        const input = findByTestAttr(wrapper, "testInputUsername");
        const input2 = findByTestAttr(wrapper, "testInputPassword");
      
        input.simulate("changeText", "usertest2@yahoo.com");
        input2.simulate("changeText", "123456");

        const button = findByTestAttr(wrapper, "press-button");
        button.simulate("press");
        wrapper.update();

        jest.spyOn(instance, 'LoginPress');
        expect(firebase.auth().signInWithEmailAndPassword).toBeCalledWith("usertest2@yahoo.com","123456");
    });
  });


