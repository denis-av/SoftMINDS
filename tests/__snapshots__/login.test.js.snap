// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.3,
          "flexDirection": "row",
          "marginTop": 70,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": "33%",
            "width": "33%",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "bold-font",
            "fontSize": 59,
            "marginLeft": "-13%",
            "marginTop": "17%",
          }
        }
      >
        SoftMINDS
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.3,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0.33,
            "justifyContent": "center",
            "marginHorizontal": "10%",
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "white",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 2,
              "flexDirection": "row",
              "height": "50%",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": "45%",
                "marginLeft": "-17%",
                "width": "45%",
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Email"
            placeholderTextColor="white"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "white",
                "fontFamily": "bold-font",
                "fontSize": 19,
                "marginLeft": "-11%",
              }
            }
            test-id="testInputUsername"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 0.33,
            "justifyContent": "center",
            "marginHorizontal": "10%",
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "white",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 2,
              "flexDirection": "row",
              "height": "50%",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": "60%",
                "marginLeft": "-24%",
                "width": "60%",
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="white"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "white",
                "fontFamily": "bold-font",
                "fontSize": 19,
                "marginLeft": "-19%",
              }
            }
            test-id="testInputPassword"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F3A416",
            "borderRadius": 30,
            "height": 47,
            "justifyContent": "center",
            "marginHorizontal": "25%",
            "width": "50%",
          }
        }
        test-id="press-button"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "bold-font",
              "fontSize": 17,
            }
          }
        >
          LOGIN
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.1,
          "justifyContent": "center",
          "marginHorizontal": "25%",
          "width": "50%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "normal-font",
              "fontSize": 16,
            }
          }
        >
          Don't have an account?
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "bold-font",
              "fontSize": 17,
            }
          }
        >
          REGISTER
        </Text>
      </View>
    </View>
  </View>
</View>
`;
